qcodes/instrument/base.py

```diff
@@ -30,30 +30,32 @@ class InstrumentBase(Metadatable, DelegateAttributes):
             attaching it to a Station.
         metadata: additional static metadata to add to this
             instrument's JSON snapshot.
+    """
 
-
-    Attributes:
-        name (str): An identifier for this instrument, particularly for
-            attaching it to a Station.
-
-        parameters (Dict[Parameter]): All the parameters supported by this
-            instrument. Usually populated via ``add_parameter``.
-
-        functions (Dict[Function]): All the functions supported by this
-            instrument. Usually populated via ``add_function``.
-        submodules (Dict[Metadatable]): All the submodules of this instrument
-            such as channel lists or logical groupings of parameters.
-            Usually populated via ``add_submodule``.
+    functions: Dict[str, Function] = {}
```

Comment:
i wouldn't want `functions` and `submodules` to be perceived as class attributes as opposed to instance attributes. So i'd prefer that they are documented the same ways as `parameters` - in the `__init__`. Alternatively, perhaps, it's possible to declare the attribute at the class level (with typing and docstring) but not give it any value?

Reply:
Yes, this pr should really have been draft. It's meant to show different options. They should all be done the same way